validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java

```diff
@@ -24,13 +31,28 @@
 
 public class KeyManager {
 
+  public enum ValidatorKeystoreDirectories {
+    KEYSTORES,
+    KEYSTORE_PASSWORDS
+  }
   /**
    * Get a listing of active validator keys
    *
    * @return a list of active validators
    */
-  List<ActiveValidator> getActiveValidatorKeys() {
-    throw new NotImplementedException("getActiveValidatorKeys not implemented yet");
+  public List<ActiveValidator> getActiveValidatorKeys() {
+    List<ActiveValidator> activeValidatorList = new ArrayList<>();
+    Set<BLSPublicKey> blsPublicKeySet = validatorLoader.getOwnedValidators().getPublicKeys();
```

Comment:
I think the way I'd go is we set an attribute on Validator (returned by ownedValidators) that the key is read-only by default, and when we start adding keys, we can set that readonly to false...
then this readonly attribute comes directly off of the getOwnedValidators list. I'm not a huge fan of crawling the filesystem constantly because it'd be an attack vector potentially

Reply:
I think the key here will be when 'whatever' reads the keys at startup and reload, they will definitely need to know about this new path and when reading from that new path, they'll set readonly to false - so if we're doing listings from the new path in a test, this logic may need to be in that read location...

Reply:
yes, currently the validator list is loaded/read by the `ValidatorLoader` from different ValidatorSources, so when we implement the load keystore (from the new path) function we can validate it.